### ISSUE TITLE
[BUG] Error: Failed to query available provider packages

### ISSUE THREAD

**Issue by scheung38** [rancher/terraform-provider-rke#270]
Steps to produce:
terraform init

Warning: Interpolation-only expressions are deprecated

 on rke.tf line 64, in resource "rke_cluster" "cluster":
 64:   addons = "${data.template_file.addons.rendered}"

Error: Failed to query available provider packages

Could not retrieve the list of available versions for provider
hashicorp/digitalocean: provider registry registry.terraform.io does not have
a provider named registry.terraform.io/hashicorp/digitalocean

Error: Failed to query available provider packages

Could not retrieve the list of available versions for provider hashicorp/rke:
provider registry registry.terraform.io does not have a provider named
registry.terraform.io/hashicorp/rke

(.venv) ➜  digitalocean git:(master) terraform --version
Terraform v0.14.2

Mac Big Sur: 11.1

**Comment by munro** [comment]
I'm getting a similar error

I have
```terraform
variable "do_token" {}

provider "digitalocean" {
  token = var.do_token
}
```

then when I run

```shell
terraform init
```

I get

```
Initializing the backend...

Initializing provider plugins...
- Finding latest version of hashicorp/digitalocean...

Error: Failed to query available provider packages

Could not retrieve the list of available versions for provider
hashicorp/digitalocean: provider registry registry.terraform.io does not have
a provider named registry.terraform.io/hashicorp/digitalocean

If you have just upgraded directly from Terraform v0.12 to Terraform v0.14
then please upgrade to Terraform v0.13 first and follow the upgrade guide for
that release, which might help you address this problem.

Did you intend to use digitalocean/digitalocean? If so, you must specify that
source address in each module which requires that provider. To see which
modules are currently depending on hashicorp/digitalocean, run the following
command:
    terraform providers
```

**Comment by munro** [comment]
Oh, I figured it out.  @scheung38  try adding this

```terraform

terraform {
  required_providers {
    digitalocean = {
      source = "digitalocean/digitalocean"
    }
  }
}
```

**Comment by aireddy73** [comment]
Error: Failed to query available provider packages

Could not retrieve the list of available versions for provider
hashicorp/digitalocean: provider registry registry.terraform.io does not have
a provider named registry.terraform.io/hashicorp/digitalocean

If you have just upgraded directly from Terraform v0.12 to Terraform v0.14
then please upgrade to Terraform v0.13 first and follow the upgrade guide for
that release, which might help you address this problem.

Did you intend to use digitalocean/digitalocean? If so, you must specify that
source address in each module which requires that provider. To see which
modules are currently depending on hashicorp/digitalocean, run the following
command:
    terraform providers 


I am using terraform 14.5 on Windows10

**Comment by aireddy73** [comment]
It worked after adding below mentioned provider block

terraform {
  required_providers {
    digitalocean = {
      source = "digitalocean/digitalocean"
    }
  }
}

**Comment by asmasud** [comment]
> Oh, I figured it out. @scheung38 try adding this
> 
> ```terraform
> terraform {
>   required_providers {
>     digitalocean = {
>       source = "digitalocean/digitalocean"
>     }
>   }
> }
> ```

**Comment by asmasud** [comment]
Thanks it worked for me too

**Comment by rawmind0** [comment]
Provider examples were created to use terraform 0.12.x. They has not been updated yet, to use tf 0.13.x and `registry.terraform.io`.  We've published a guide to update this provider to tf 0.13.x, `https://registry.terraform.io/providers/rancher/rke/latest/docs/guides/upgrade_to_0.13` . 

As mentioned, this will make the example works
```terraform
terraform {
  required_providers {
    digitalocean = {
      source = "digitalocean/digitalocean"
      version = "X.Y.Z"
    }
  }
}
```

Anyway, PR https://github.com/rancher/terraform-provider-rke/pull/272 has been submitted to update DO examples. Once reviewed, it would be merged.

**Comment by jadamcraig** [comment]
@rawmind0 , Yes, now that you mention it, the issue originally reported by @scheung38 above is actually what started me down the path of working on PR #272 .

I am currently working on a similar example template for Linode.  Also, since I'm hearing rumours that RancherOS may soon be abandoned, I'll plan to review the DO example again once DigitalOcean themselves decide which container OS to include in their staff-maintained images.  Presently, RancherOS is the only option there, though.

**Comment by rawmind0** [comment]
@jadamcraig , thanks for working on that PR.

RancherOS may be fine by the moment, but it's arriving to EOL soon, https://github.com/rancher/os/issues/3000#issuecomment-630511978

**Comment by manuelemeka** [comment]
terraform {
  required_providers {
    digitalocean = {
      source = "digitalocean/digitalocean"
    }
  }
}

**Comment by air3ijai** [comment]
In our case the issue was related to the lack of the `required_providers` block in the module.

Root module - `versions.tf`
```terraform
# Terraform settings
terraform {
  required_version = "~> 1.6"
  required_providers {
    digitalocean = {
      source  = "digitalocean/digitalocean"
      version = "~> 2.0"
    }
  }
}
```

Child module - `modules/doks/versions.tf`
```terraform
# Terraform settings
terraform {
  required_providers {
    digitalocean = {
      source  = "digitalocean/digitalocean"
      version = "~> 2.0"
    }
  }
}
```

Reference 
1. [Using a non hashicorp provider in a module](https://discuss.hashicorp.com/t/using-a-non-hashicorp-provider-in-a-module/21841)
2. [Why doesn’t required_providers in a child module inherit declarations from the parent?](https://discuss.hashicorp.com/t/why-doesnt-required-providers-in-a-child-module-inherit-declarations-from-the-parent/41493)
